src/Bicep.IO/Utils/TgzFileHandle.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Frozen;
+using System.Formats.Tar;
+using System.IO.Compression;
+using Bicep.IO.Abstraction;
+
+namespace Bicep.IO.Utils
+{
+    public class TgzFileHandle
+    {
+        private readonly IFileHandle fileHandle;
+
+        public TgzFileHandle(IFileHandle fileHandle)
+        {
+            this.fileHandle = fileHandle;
+        }
+
+        public bool Exists() => this.fileHandle.Exists();
+
+        public FrozenDictionary<string, string> Extract()
+        {
+            if (!this.Exists())
+            {
+                throw new InvalidOperationException("The file does not exist.");
```

Comment:
Include path in error?

Reply:
Good point. Updated the error message to include the file URI.